### ISSUE TITLE
Add onWarpSyncDone callback to smoldot client config

### ISSUE THREAD

**Issue by tomaka** [paritytech/smoldot#720]
The callback should be called after we have finished GrandPa warp syncing.

This is useful for UX purposes.

**Comment by tomaka** [comment]
I started implementing this and realized that, rather than adding a special code path for GrandPa warp sync, we could just detect the first ever best block update, since it always happens at the end of the warp syncing.

But then I realized that it could also be done by subscribing to new best blocks through the JSON-RPC endpoint.
However we can't entirely delegate this entirely to API users because of a race condition: the warp syncing might already be done when they send the subscription.
Additionally, even if we send a JSON-RPC request at initialization, these requests are processed asynchronously, and the request might be processed after the warp syncing has finished.

In other words, we still need some callback system, but it can simply be implemented in `lib.rs` by subscribing to new best blocks.

**Comment by tomaka** [comment]
After more thinking:

What the UI on top should do is call `system_health` and consider itself as connected if `(health.peers > 0 || !health.shouldHavePeers) && !health.isSyncing`.

Even better, the UI should also subscribe to best blocks and call `system_health` whenever a new best block arrives.
This should guarantee a minimum delay between the end of the syncing and the moment the UI realized we're connected.

**Comment by tomaka** [comment]
I'm going to close this issue as unnecessary.
